src/librc/librc-misc.c

```diff
@@ -443,3 +444,18 @@ rc_conf_value(const char *setting)
 
 	return rc_config_value(rc_conf, setting);
 }
+
+void rc_export_variable(const char *name, const char *value) {
+	char *envpath;
+	FILE *envfile;
+
+	xasprintf(&envpath, "%s/env", rc_svcdir());
+
+	if (!(envfile = fopen(envpath, "a+")))
+		return;
+	if (flock(fileno(envfile), LOCK_EX) == -1)
+		return;
+	/* TODO: Remove repeated entries, before inserting a new one. */
+	fprintf(envfile, "export %s=%s\n", name, value);
+	fclose(envfile);
```

Comment:
Should not assume that the writes succeeded.
```suggestion
        int err = fflush(envfile);
	if (fclose(envfile) != 0 || err)
	    // log
```